wrappers/desktop.go

```diff
@@ -171,6 +171,11 @@ func sanitizeDesktopFile(s *snap.Info, desktopFile string, rawcontent []byte) []
 		newContent.Grow(len(bline) + 1)
 		newContent.Write(bline)
 		newContent.WriteByte('\n')
+
+		// insert snap name
+		if bytes.Equal(bline, []byte("[Desktop Entry]")) {
+			newContent.Write([]byte("X-SnapName=" + s.InstanceName() + "\n"))
```

Comment:
As long as people don't include `X-SnapName=` in their desktop files themselves.

Reply:
The desktop files are validated and this field is not allowed.